packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,297 @@
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent?.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+        return node.parent.property.name;
+      }
+      return undefined;
+    }
+
+    function isLoadFunction(node: TSESTree.Node): boolean {
+      return (node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && node.parent.property.name === "load");
+    }
+
+    function callsGetAPIFunction(node: TSESTree.Identifier): boolean {
+      return getFunctions.has(node.name);
+    }
+
+    function isVariableDeclaration(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator) {
+          return true;
+      }
+      return false;
+    }
+
+    function isGetVariableDeclaration(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator
+        && node.parent.init?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.init.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.init.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.parent.init.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function isAssignmentExpression(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.AssignmentExpression) {
+          return true;
+      }
+      return false;
+    }
+
+    function isGetAssignmentExpression(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.AssignmentExpression
+        && node.parent.right.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.right.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.right.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.parent.right.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function getLoadedPropertyName(node: TSESTree.Node): string {
+      if(node.parent?.parent?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.parent.arguments[0].type === TSESTree.AST_NODE_TYPES.Literal) {
+          return node.parent.parent.arguments[0].value as string;
+        }
+      return "error in getLoadedPropertyName";
+    }
+
+    function findLoadBeforeRead(scope: Scope) {
+      scope.variables.forEach((variable: Variable) => {
+        let loadLocation: Map <string, number> = new Map<string, number>();
+        let getFound: boolean = false;
+        variable.references.forEach((reference: Reference) => {
+          const node: TSESTree.Node = reference.identifier;
+          if (isVariableDeclaration(node)
```

Comment:
Could probably also check the "getFound" here and not do the other checks if it was already seen.

Reply:
It is determining if the return will happen or not across different references, so it cannot go inside the if.
The get found will have the same value across different iterations.

Reply:
So maybe I'm a little confused about the reason for "getFound".  It is defined outside the forEach, but it is getting set each time you go in.  You are are returning right after you set it to true, but then returning if "!getFound" (which should be every case that isn't a get statement).  Reading it this time around makes be wonder how the rule is finding an error at all.  Seems like there is something I'm missing.